docs/configuration/providers.md

```diff
@@ -0,0 +1,38 @@
+## Providers
+
+### GitHub Enterprise
+
+1. Set the following configuration from your GitHub instance on the values.yaml file:
+
+    ```
+    gitHubEnterprise:
+        hostname: "CHANGE_ME"
+        protocol: "CHANGE_ME"
+        isPrivateMode: "CHANGE_ME"
+        port: "CHANGE_ME"                      # This is an optional field
+        disableSSL: "CHANGE_ME"                # This is an optional field
+    ```
+
+### GitLab Enterprise
+
+1. Set the following configurations from your GitLab instance on the values.yaml file:
+
+    ```
+    gitLabEnterprise:
+        hostname: "CHANGE_ME"
+        protocol: "CHANGE_ME"
+    ```
+
+### Bitbucket Enterprise
+
```

Comment:
Can you add here the gifs from this page that might be relevant for me to complete the setup?
https://support.codacy.com/hc/en-us/articles/207280509-Configuring-Bitbucket-Server-from-Stash-with-Codacy-Self-hosted